### ISSUE TITLE
Can't build release 5.13

### ISSUE THREAD

**Issue by hhoangna** [TelegramMessenger/Telegram-iOS#289]
I build based on Telegram's document: https://core.telegram.org/reproducible-builds#reproducible-builds-for-ios
And i got that error in step 7. Someone can help me, pls

1. ERROR in /Users/hhumorous/buck_source/buck/src/com/facebook/buck/jvm/java/abi/JarReader.java (at line 71)
    [javac] 	fileSystem = FileSystems.newFileSystem(jarPath, null);
    [javac] 	                         ^^^^^^^^^^^^^
    [javac] The method newFileSystem(Path, ClassLoader) is ambiguous for the type FileSystems
    [javac] ----------
    [javac] ----------
    [javac] 2. ERROR in /Users/hhumorous/buck_source/buck/src/com/facebook/buck/jvm/java/abi/java11/SourceVersionUtils.java (at line 27)
    [javac] 	switch (version) {
    [javac] 	        ^^^^^^^
    [javac] The enum constant RELEASE_12 should have a corresponding case label in this enum switch on SourceVersion. To suppress this problem, add a comment //$CASES-OMITTED$ on the line above the 'default:'
    [javac] ----------
    [javac] 3. ERROR in /Users/hhumorous/buck_source/buck/src/com/facebook/buck/jvm/java/abi/java11/SourceVersionUtils.java (at line 27)
    [javac] 	switch (version) {
    [javac] 	        ^^^^^^^
    [javac] The enum constant RELEASE_13 should have a corresponding case label in this enum switch on SourceVersion. To suppress this problem, add a comment //$CASES-OMITTED$ on the line above the 'default:'
    [javac] ----------
    [javac] ----------
    [javac] 4. ERROR in /Users/hhumorous/buck_source/buck/src/com/facebook/buck/util/unarchive/Unzip.java (at line 320)
    [javac] 	try (FileSystem zipFs = FileSystems.newFileSystem(archiveAbsolutePath, null)) {
    [javac] 	                                    ^^^^^^^^^^^^^
    [javac] The method newFileSystem(Path, ClassLoader) is ambiguous for the type FileSystems
    [javac] ----------
    [javac] 4 problems (4 errors)

BUILD FAILED
/Users/hhumorous/buck_source/buck/build.xml:588: The following error occurred while executing this line:
/Users/hhumorous/buck_source/buck/build.xml:530: Compile failed; see the compiler error output for details.

**Comment by victoriaqb** [comment]
same issue

**Comment by Qudratillo** [comment]
Did you set JAVA_HOME? You must have java 8. I recommend you to use jenv if you have multiple java environments. `jenv  enable-plugin export` helps to set JAVA_HOME variable.

**Comment by Qudratillo** [comment]
Did you set JAVA_HOME? You must have java 8. I recommend you to use jenv if you have multiple java environments. `jenv  enable-plugin export` helps to set JAVA_HOME variable.

**Comment by marv3lls** [comment]
# TL;DR to get it working at bottom

> Did you set JAVA_HOME? You must have java 8. I recommend you to use jenv if you have multiple java environments. `jenv enable-plugin export` helps to set JAVA_HOME variable.

`java_home` returns `/Library/Java/JavaVirtualMachines/openjdk-14.0.2.jdk/Contents/Home`

I set `JAVA_HOME="/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/"`

and re-ran `./prepare_buck_source.sh $HOME/buck_source`

And get the same error.

I `brew install jenv` and ran that command. Failed again. Added adopt to jenv like so:

```
jenv add /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/
openjdk64-1.8.0.265 added
1.8.0.265 added
1.8 added
```
and
```
jenv add /Library/Java/JavaVirtualMachines/openjdk-14.0.2.jdk/Contents/Home
openjdk64-14.0.2 added
14.0.2 added
14.0 added
14 added
```
Then set `jenv global 1.8`

And re-ran `./prepare_buck_source.sh $HOME/buck_source` to build successfully, only to get:

```
BUILD SUCCESSFUL
Total time: 2 minutes 34 seconds
+ PATH===REDACTED===
+ ./bin/buck build --show-output buck
Starting new Buck daemon...
Buck encountered an internal error
com.google.common.util.concurrent.ExecutionError: java.lang.UnsupportedClassVersionError: com/facebook/buck/remoteexecution/proto/RESessionID has been compiled by a more recent version of the Java Runtime (class file version 58.0), this version of the Java Runtime only recognizes class file versions up to 52.0
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2216)
	at com.google.common.cache.LocalCache.get(LocalCache.java:4147)
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4151)
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5140)
	at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:5146)
	at com.facebook.buck.core.config.ConfigViewCache.getView(ConfigViewCache.java:45)
	at com.facebook.buck.core.config.BuckConfig.getView(BuckConfig.java:101)
	at com.facebook.buck.cli.MainRunner.isRemoteExecutionAutoEnabled(MainRunner.java:1702)
	at com.facebook.buck.cli.MainRunner.isRemoteExecutionBuild(MainRunner.java:1713)
	at com.facebook.buck.cli.MainRunner.runMainWithExitCode(MainRunner.java:741)
	at com.facebook.buck.cli.MainRunner.runMainThenExit(MainRunner.java:478)
	at com.facebook.buck.cli.MainWithNailgun.nailMain(MainWithNailgun.java:76)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.facebook.nailgun.NGSession.runImpl(NGSession.java:313)
	at com.facebook.nailgun.NGSession.run(NGSession.java:199)
Caused by: java.lang.UnsupportedClassVersionError: com/facebook/buck/remoteexecution/proto/RESessionID has been compiled by a more recent version of the Java Runtime (class file version 58.0), this version of the Java Runtime only recognizes class file versions up to 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:468)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
	at java.lang.Class.privateGetMethodRecursive(Class.java:3048)
	at java.lang.Class.getMethod0(Class.java:3018)
	at java.lang.Class.getMethod(Class.java:1784)
	at com.facebook.buck.core.config.ConfigViewCache$ConfigViewCacheLoader.load(ConfigViewCache.java:66)
	at com.facebook.buck.core.config.ConfigViewCache$ConfigViewCacheLoader.load(ConfigViewCache.java:1)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3708)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2416)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2299)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2212)
	... 17 more
```

## TL;DR:

- `brew install jenv`
- Restart shell, adding necessary lines to `.profile` or `.zshrc` the last step told you to.
- `jenv add /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/`
- `jenv global 1.8`
- `./prepare_buck_source.sh $HOME/buck_source`
- Still not continuing past initial build to run buck properly?

**Comment by marv3lls** [comment]
Continuing `/Users/user/buck_source/buck/bin/buck build --show-output buck`

Outputs the same error again, even when changing what jdk jenv uses

**Comment by ali-fareed** [comment]
Please see the updated build instructions in README.md.
